src/stateManager/StateManager.ts

```diff
@@ -0,0 +1,778 @@
+            console.log("tryExecuteFromQueue - executing");
+            if (await this.onSignedBlock(signedBlock)) {
+                break;
+            }
+        }
+    }
+    private async tryConfirmFromQueue(): Promise<void> {
+        //TODO! race condition and skipping a txCount
+        let confirmations = this.agreementManager.tryDequeueConfirmations(
+            Number(this.getForkCnt()),
+            Number(this.getNextTransactionCnt())
+        );
+
+        for (const confirmation of confirmations) {
+            if (
+                await this.onBlockConfirmation(
+                    confirmation.originalSignedBlock,
+                    confirmation.confirmationSignature as string
+                )
+            ) {
+                break;
+            }
+        }
+    }
+    /**
+     * Triggered by the On-chain Event Listener when a new state is set on-chain
+     * @param encodedState - Encoded state of the state machine
+     * @param _forkCnt - new fork count
+     * @param _timestamp - on-chain timestamp
+     */
+    public async setState(
+        encodedState: string,
+        _forkCnt: BigNumberish,
+        _timestamp: BigNumberish
+    ): Promise<void> {
+        console.log("StateManager - SetState", _forkCnt, _timestamp);
+        await this.stateMachine.setState(encodedState);
+        this.agreementManager.newFork(
+            encodedState,
+            await this.stateMachine.getParticipants(),
+            Number(_forkCnt),
+            Number(_timestamp)
+        );
+
+        //Try timeout next participant
+        this.p2pEventHooks.onSetState?.();
+        return this.onSuccessCommon();
+    }
+
+    // Passes the signedBlock through a verification pipeline and returns an execution flag based on the outcome
+    public async onSignedBlock(
+        signedBlock: SignedBlockStruct
+    ): Promise<ExecutionFlags> {
+        // Default everything to SUCCESS + no AgreementFlag
+        let finalExecutionFlag: ExecutionFlags = ExecutionFlags.SUCCESS;
+        let finalAgreementFlag: AgreementFlag | undefined = undefined;
+
+        try {
+            await this.mutex.lock();
+
+            const block = EvmUtils.decodeBlock(signedBlock.encodedBlock);
+
+            const context: ValidationContext = {
+                stateManager: this,
+                signedBlock,
+                block
+            };
+
+            const validators: ValidationStep<ValidationContext>[] = [
+                checkManagerReadiness,
+                checkBlockValidity,
+                checkBlockForkStatus,
+                checkForkDisputeStatus,
+                checkDuplicateBlock,
+                checkBlockIsFuture,
+                checkParticipantInFork,
+                checkPastBlockCurrentFork,
+                checkBlockTimestamp,
+                checkEnoughTimeSubjective,
+                checkCorrectBlockProducer,
+                verifyStateTransition
+            ];
```

Comment:
This is my favorite part of this PR - clearly shows the checks - maybe not the best naming on some, but we can adjust that later. 

We have pipelines and checks everywhere, but the current pipeline logic (as used here) only works due to the way how the underlying pipeline logic is structured - chain on 'success' and pas 'final flags' to some function (processConfirmation/Execution decisions)

Most pipeline steps are 1 or 2 lines and not sure how much a 'step' could be reused and the only benefit to this approach vs just refactoring the validation/confirmation logic in another file into smaller function calls, is that we don't explicitly check 'if flag == success' all the time

Overall I like it, but we'll see how practical it will turnout in the end